lib/api/core/models/repositories/profileRepository.js

```diff
@@ -271,6 +271,11 @@ class ProfileRepository extends Repository {
 
     return profile.validateDefinition()
       .then(() => {
+        if (profile._id === 'anonymous' && profile.policies.map(policy => policy.roleId).indexOf('anonymous') === -1) {
```

Comment:
Same comment as @scottinet. This test can't be before `profile.validateDefinition()`?